json-schema-converter/src/main/java/io/confluent/connect/json/JsonSchemaData.java

```diff
@@ -219,7 +222,19 @@ private static boolean isInstanceOfSchemaTypeForSimpleSchema(Schema fieldSchema,
       case MAP:
         return value.isObject() || value.isArray();
       case STRUCT:
-        return value.isObject();
+        if (value.isObject()) {
+          Set<String> schemaFields = fieldSchema.fields()
+              .stream()
+              .map(Field::name)
+              .collect(Collectors.toSet());
+          Set<String> objectFields = new HashSet<>();
+          for (Iterator<Entry<String, JsonNode>> iter = value.fields(); iter.hasNext(); ) {
+            objectFields.add(iter.next().getKey());
+          }
+          return schemaFields.containsAll(objectFields);
```

Comment:
It's maybe a bit more complicated since:
- all required fields should be there
- all non-required fields _can_ be there

The match with most non-required fields matching should probably be the winner.

Reply:
Thanks for the feedback, @frankkoornstra.  Unfortunately, this method does not have access to the original JSON Schema, only the Connect struct schema.  Anyway, I improved it a little bit to find the Connect struct schema with the largest intersection.  It should be noted that some JSON Schemas allow for properties not defined in the schema, when `"additionalProperties=true"`, which is the default.